Data Structures/HashMap/CustomHashmap.java

```diff
@@ -0,0 +1,294 @@
+        }
+        // If the hash table slot is not empty, search for the key in the linked list
+        else{
+            Entry current = table[hash];
+            Entry previous = null;
+            while (current != null){
+                // If the key is found, remove the corresponding key-value pair from the linked list
+                if (current.key == key){
+                    if (previous == null){
+                        table[hash] = table[hash].next;
+                        size -= 1;
+                        return true;
+                    }
+                    else{
+                        previous.next = current.next;
+                        size -= 1;
+                        return true;
+                    }
+                }
+                previous = current;
+                current = current.next;
+            }
+            return false;
+        }
+    }
+
+    // Method to get the value associated with a given key in the hash map
+    int getValue(int key){
+        // Calculate the hash code for the key
+        int hash = getHash(key);
+        Entry current = table[hash];
+        // If the hash table slot is empty, the key is not in the hash map
+        if (current == null){
+            return -1;
+        }
+        // If the hash table slot is not empty, search for the key in the linked list
+        else{
+            while (current != null){
+                if (current.key == key){
+                    return current.value;
+                }
+                current = current.next;
+            }
+        }
+        return -1;
+    }
+
+    // Method to check if a given key is in the hash map
+    boolean search(int key){
+        
+
+    
+        int hash= getHash(key);
+
+        if (table[hash]==null)
+        {  
+            return false;
+        }
+
+        else{
+
+
+            Entry current= table[hash];
+
+            while(current!= null){
+
+                if(current.key==key){
+                    return true;
+                }
+
+                current= current.next;
+
+            }
+
+        }
+
+        return false;
+
+
+
+
+    }
+
+    // Method to display the adjacency matrix
+
+    void display()
+    {
+
+        for(int i=0; i<capacity;i++)
+        {
+
+
+
+            if (table[i]!=null)
+            {
+
+                System.out.print(i + ":");
+
+                Entry ptr= table[i];
+
+                while(ptr!= null)
+                {
+                    System.out.print("("+ ptr.key + ","+ ptr.value + ")" + " ");
+
+                    ptr= ptr.next;
+
+                }
+
+                System.out.println();
+
+            }
+            else{
+
+                System.out.println(i);
+            }
+
+
+
+        }
+
+
+    }
+
+
+    int getHash(int key){
+
+        int hash= key%capacity;
+        return hash;
+
+    }
+
+
+    public static void main(String[] args) {
+
+        try (Scanner input = new Scanner(System.in)) {
```

Comment:
Using this try block will terminate the program in case an exception occurred. The program should get terminated only by menu-driven choice by the user. Please use it properly.